CRM/Utils/PDF/Utils.php

```diff
@@ -127,6 +127,14 @@ public static function html2pdf(&$text, $fileName = 'civicrm.pdf', $output = FAL
     </div>
   </body>
 </html>";
+
+    //output HTML if needed
```

Comment:
So when I see patterns like this it says to me that the portion of code that precedes this should be extracted into it's own function - (e.g getFullProcessedHtmll) and then we call that function from here & from the calling function.

It's not terribly clear to me that function even belongs on this class - ie. it seems kinda generic - although I'd be Ok leaving it here for now

Reply:
(also the resulting function would actually be unit testable - yay)

Reply:
Ok I'll have a go at extracting the code and splitting that into its own function.